src/ngraph/runtime/cpu/builder/quantized_dot.cpp

```diff
@@ -114,70 +115,50 @@ namespace ngraph
-                            node);
-                    size_t ip_index = mkldnn_emitter->inner_product_forward_init(false);
-                    auto& deps = mkldnn_emitter->get_primitive_deps(ip_index);
+                auto arg0_buffer_index = external_function->get_buffer_index(args[0].get_name());
+                auto arg1_buffer_index = external_function->get_buffer_index(args[1].get_name());
+                auto arg2_buffer_index = external_function->get_buffer_index(args[2].get_name());
+                auto out0_buffer_index = external_function->get_buffer_index(out[0].get_name());
 
-                    auto functor = [&,
-                                    scales_size,
-                                    ip_desc,
-                                    ip_attr,
-                                    deps,
-                                    ip_index,
-                                    arg0_buffer_index,
-                                    arg1_buffer_index,
-                                    arg2_buffer_index,
-                                    out0_buffer_index](CPURuntimeContext* ctx,
-                                                       CPUExecutionContext* ectx) mutable {
-                        if (ctx->first_iteration)
-                        {
-                            vector<float> dyn_scales;
-                            dyn_scales.assign(
-                                static_cast<float*>(ctx->buffer_data[arg2_buffer_index]),
-                                static_cast<float*>(ctx->buffer_data[arg2_buffer_index]) +
-                                    scales_size);
-                            ip_attr.set_output_scales(0, dyn_scales);
-                            mkldnn_emitter->build_inner_product_forward<false>(
-                                ctx->mkldnn_primitives,
-                                ip_desc,
-                                ip_attr,
-                                executor::global_cpu_engine,
-                                deps,
-                                ip_index);
-                        }
-                        cpu::mkldnn_utils::set_memory_ptr(
-                            ctx, deps[0], ctx->buffer_data[arg0_buffer_index]);
-                        cpu::mkldnn_utils::set_memory_ptr(
-                            ctx, deps[1], ctx->buffer_data[arg1_buffer_index]);
-                        cpu::mkldnn_utils::set_memory_ptr(
-                            ctx, deps[2], ctx->buffer_data[out0_buffer_index]);
-                        cpu::mkldnn_utils::mkldnn_invoke_primitive(ctx, ip_index);
-                    };
-                    functors.emplace_back(functor);
-                }
-                else
-                {
-                    throw ngraph_error("unsupported parameters for QuantizedDot via DEX");
-                }
+                auto scales_size = shape_size(args[2].get_shape());
```

Comment:
check that this is equal to '1'.  Even better, do this in QuantizeDot's validate_and_infer_types method

Reply:
not sure, but we may need to allow for non-scalar scales in this op at some later date.  could we not just reject this op in cpu assignment is scale is a non-scalar.

Reply:
Current ref kernel only supports a scalar scale though?